sdk/src/main/java/io/dapr/client/AbstractDaprClient.java

```diff
@@ -392,6 +396,34 @@ public <T> Mono<QueryStateResponse<T>> queryState(QueryStateRequest request, Cla
     return this.queryState(request, TypeRef.get(clazz));
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T> Mono<BulkPublishResponse<T>> publishEvents(String pubsubName, String topicName, List<T> events,
+                                                     String contentType) {
+    return publishEvents(pubsubName, topicName, events, contentType, null);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T> Mono<BulkPublishResponse<T>> publishEvents(String pubsubName, String topicName, List<T> events,
+                                                     String contentType, Map<String, String> requestMetadata) {
+    BulkPublishRequest<T> request = new BulkPublishRequest<>(pubsubName, topicName);
+    if (events == null || events.size() == 0) {
+      throw new IllegalArgumentException("list of events cannot be null or empty");
+    }
+    List<BulkPublishEntry<T>> entries = new ArrayList<>();
+    for (int i = 0; i < events.size(); i++) {
+      // entryID field is generated based on order of events in the request
+      entries.add(new BulkPublishEntry<>("" + (i + 1), events.get(i), contentType, null));
```

Comment:
I think starting from zero is more idiomatic to Java developers.

Reply:
changed